modules/core/common/createReduxStore.ts

```diff
@@ -8,11 +8,63 @@ export const getStoreReducer = (reducers: any) =>
     ...reducers
   });
 
-const createReduxStore = (reducers: Reducer, initialState: DeepPartial<any>, routerMiddleware?: Middleware): Store => {
+const requestMiddleware: Middleware = _state => next => action => {
+  const { types, APICall, ...rest } = action;
+  if (!types) {
+    return next(action);
+  }
+
+  const { REQUEST, SUCCESS, FAIL } = types;
+
+  next({ type: REQUEST || null, ...rest });
+
+  const handleAPICall = async () => {
+    try {
+      const result = await APICall();
+      const data = result && result.data;
+      if (data.errors) {
```

Comment:
According to this `if` statement the error can be sent via either data in response with status `200` or response with any other statuses. I don't think it makes any sense and we should have just one flow for sending errors.